build/dsc/templates/dsc_type.rb.erb

```diff
@@ -74,28 +74,24 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   <%  end -%>
   end
 
-  newproperty(:dscmeta_resource_friendly_name) do
+  newparam(:dscmeta_resource_friendly_name) do
     desc "A read-only value that is the DSC Resource Friendly Name ('<%= resource.friendlyname %>')."
 
-    def retrieve
-      '<%= resource.friendlyname %>'
-    end
-
     validate do |value|
-      fail 'dscmeta_resource_friendly_name is read-only'
+      fail 'dscmeta_resource_friendly_name is read-only' if value != '<%= resource.friendlyname %>'
```

Comment:
I think you can remove the `if` ... lets make it simpler and just fail if a user tries to set it at all (which is the method we use in AWS module - i.e. https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet/type/ec2_instance.rb)